internal/k6runner/k6runner.go

```diff
@@ -337,17 +337,32 @@ func (r HttpRunner) WithLogger(logger *zerolog.Logger) Runner {
 var ErrUnexpectedStatus = errors.New("unexpected status code")
 
 func (r HttpRunner) Run(ctx context.Context, script []byte) (*RunResponse, error) {
-	req, err := json.Marshal(&RunRequest{
+	k6Timeout := getTimeout(ctx)
+
+	reqBody, err := json.Marshal(&RunRequest{
 		Script: script,
 		Settings: Settings{
-			Timeout: getTimeout(ctx).Milliseconds(),
+			Timeout: k6Timeout.Milliseconds(),
 		},
 	})
 	if err != nil {
 		return nil, fmt.Errorf("running script: %w", err)
 	}
 
-	resp, err := http.Post(r.url, "application/json", bytes.NewBuffer(req))
+	// The context above carries the check timeout, which will be eventually passed to k6 by the runner at the other end
+	// of this request. To account for network overhead, we add a second of grace time to the script timeout to form
+	// the timeout of this request.
+	reqCtx, cancel := context.WithTimeout(ctx, k6Timeout+time.Second)
```

Comment:
does this actually work? The parent context `ctx` will timeout first, causing the child context to timeout, too, and ignore the extra second...

Reply:
I think you're right, I was thinking "if this is not timing out before it won't now" but that does not make any sense...

Reply:
I've worked around this by creating a different context entirely: https://github.com/grafana/synthetic-monitoring-agent/pull/715/commits/409666a0f1035b4a3592c8379618d6101ef5133e